sentinel.py

```diff
@@ -58,31 +77,37 @@ def create(name, mro=(object,), extra_methods={}, *args, **kwargs):
     instantiating base classes such as a tuple.
     """
 
-    cls_dict = {}
-
-    cls_dict.update(
-        # Provide a nice, clean, self-documenting __repr__
-        __repr__=lambda self: name,
-        # Provide a copy and deepcopy implementation which simply return the
-        # same exact instance.
-        __deepcopy__=lambda self, _memo: self,
-        __copy__=lambda self: self,
-        # Provide a hook for pickling the sentinel.
-        __reduce__=lambda self: name
-    )
-
-    # If the default MRO is given, then it's safe to prevent the singleton
-    # instance from having a instance dictionary.
-    if mro == (object,):
-        cls_dict.update(__slots__=())
-
-    cls_dict.update(extra_methods)
-
-    anon_type = type(name, mro, cls_dict)
-
-    # Stack introspection -- make the singleton always belong to the module of
-    # its caller. If we don't do this, pickling using __reduce__() will fail!
-    anon_type.__module__ = get_caller_module()
-
-    # Return the singleton instance of this new, "anonymous" type.
-    return anon_type(*args, **kwargs)
+    _cls_dict = {
+        '__module__': _get_caller_module(),
+        '__repr__': lambda _: name,
+        '__reduce__': lambda _: name,
+        '__copy__': lambda self: self,
+        '__deepcopy__': lambda self, _: self
+    }
+    if mro == __mro_immortal:
+        mro = ()
+        _cls_dict['__slots__'] = ()
+    elif object in mro:
+        assert mro[-1] is object # object is ALWAYS the last type in a mro
```

Comment:
This is the reason assertions were created! 😂 

```suggestion
        assert mro[-1] is object, "object should ALWAYS be the last type in a mro"
```